sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -95,6 +95,41 @@ def test_mean_variance_axis1():
     assert_raises(TypeError, mean_variance_axis, X_lil, axis=1)
 
 
+def test_mean_variance_axisnone():
+    axis = None
+    X, _ = make_classification(5, 4, random_state=0)
+    # Sparsify the array a little bit
+    X[0, 0] = 0
+    X[2, 1] = 0
+    X[4, 3] = 0
+    X_lil = sp.lil_matrix(X)
+    X_lil[1, 0] = 0
```

Comment:
I think this was intentional to check that the code is actually dealing with "stored" zeros as opposed to zeros that are implicitly represented by non-stored entries.